de.metas.adempiere.adempiere/base/src/main/java/org/adempiere/location/geocoding/openstreetmap/NominatimOSMGeoCoordinatesProviderImpl.java

```diff
@@ -0,0 +1,128 @@
+package org.adempiere.location.geocoding.openstreetmap;
+
+import com.google.common.annotations.VisibleForTesting;
+import de.metas.util.GuavaCollectors;
+import lombok.NonNull;
+import org.adempiere.location.geocoding.GeoCoordinatesProvider;
+import org.adempiere.location.geocoding.GeographicalCoordinates;
+import org.adempiere.location.geocoding.GeoCoordinatesRequest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+import org.springframework.web.client.RestTemplate;
+
+import javax.annotation.Nonnull;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+
+/*
+ * #%L
+ * metasfresh-pharma
+ * %%
+ * Copyright (C) 2018 metas GmbH
+ * %%
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation, either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public
+ * License along with this program. If not, see
+ * <http://www.gnu.org/licenses/gpl-2.0.html>.
+ * #L%
+ */
+@Component
+public class NominatimOSMGeoCoordinatesProviderImpl implements GeoCoordinatesProvider
+{
+
```

Comment:
did you check out https://operations.osmfoundation.org/policies/nominatim/ ? 
e.g. "max 1 request per second" we might need to sleep for one set after having handeled each request..
pls make up your mind about the other requirements too

Reply:
I have not checked those. Will further research.

Reply:
Checked. 

Added rate-limiting and caching and user-agent.